### PR TITLE
Fix finding forwardRef itself

### PR DESCRIPTION
**Pull request by cyan33**
This PR is built on top of @jquense 's https://github.com/airbnb/enzyme/pull/1592

It fixes when the `wrapper.find` method cannot find the forwardRef itself. The reason this is happening is that `React.forwardRef` returns a type that is unknown to enzyme. The return type looks like this:

```js
{$$typeof: Symbol.for('react.forward_ref'), render: [Function]}
```

What we really want to search is the thing wrapped inside of `forwardRef`, so one way to do this is to call the `render` method and use the return element to re-build a predicate based on that.

### CODE REVIEW

File: packages/enzyme-adapter-utils/src/createMountWrapper.jsx [hunk 1]
```diff
@@ -3,6 +3,11 @@ import PropTypes from 'prop-types';
 
 /* eslint react/forbid-prop-types: 0 */
 
+// TODO: use react-is?
+const specialType = PropTypes.shape({
```
Comment (jquense) [inline packages/enzyme-adapter-utils/src/createMountWrapper.jsx]: should use `react-is`

File: packages/enzyme/src/selectors.js [hunk 2]
```diff
@@ -227,6 +227,12 @@ export function buildPredicate(selector) {
       if (hasUndefinedValues) {
         throw new TypeError('Enzyme::Props can’t have `undefined` values. Try using ‘findWhere()’ instead.');
       }
+      // the selector could also be a forwardRef
+      if (typeOf(selector) === ForwardRef) {
+        // re-build the predicate based on what is wrapped by forwardRef
+        // rather than the forwardRef itself
+        return buildPredicate(selector.render().props);
```
Comment (jquense) [inline packages/enzyme/src/selectors.js]: I don't think this is the right way to do this but maybe i'm misunderstanding. We should copy waht react-test-renderer does and this code should go in the react-16-adapter

Comment (aweary) [reply packages/enzyme/src/selectors.js]: Yeah the selector module is used for all adapters, so the node type should be opaque in this module.

**Comment by ljharb** [comment]
@jquense are you able to pull this PR into yours, such that we don't have to have stacked PRs?

**Comment by jquense** [comment]
I'll update my PR. I think someone actually already PR'd My PR with the correct fix

**Comment by ljharb** [comment]
(I'll leave this open and sync the two of them myself when the first one's ready to merge)

**Comment by cyan33** [comment]
Actually this PR's change is not correct as @aweary and @jquense pointed out that the change should not be in enzyme core. I've updated my fix in the comment of @jquense's PR. I'll close it for now.

**Pull request closed without merging**